integration/helpers/instance.go

```diff
@@ -967,25 +966,27 @@ func (i *TeleInstance) StartNodeAndProxy(t *testing.T, name string) (sshPort, we
 type ProxyConfig struct {
 	// Name is a proxy name
 	Name string
-	// SSHPort is SSH proxy port
-	SSHPort int
-	// WebPort is web proxy port
-	WebPort int
-	// ReverseTunnelPort is a port for reverse tunnel addresses
-	ReverseTunnelPort int
+	// SSHAddr the address the node ssh service should listen on
+	SSHAddr string
+	// WebAddr the address the web service should listen on
+	WebAddr string
+	// ReverseTunnelAddr the address the reverse proxy service should listen on
+	ReverseTunnelAddr string
 	// Disable the web service
 	DisableWebService bool
 	// Disable the web ui
 	DisableWebInterface bool
 	// Disable ALPN routing
 	DisableALPNSNIListener bool
+	// FileDescriptors holds FDs to be injected into the Teleport process
+	FileDescriptors []service.FileDescriptor
 }
 
 // StartProxy starts another Proxy Server and connects it to the cluster.
-func (i *TeleInstance) StartProxy(cfg ProxyConfig) (reversetunnel.Server, error) {
+func (i *TeleInstance) StartProxy(cfg ProxyConfig) (reversetunnel.Server, *service.TeleportProcess, error) {
```

Comment:
nit: Does returning `*service.TeleportProcess` is really needed ? right now it is used only in one please where 
 the `TeleportProcess can be also obtained by referring to `i.Nodes[len(i.Nodes)-1]` after calling `i.StartProxy`.

Reply:
Just returning it from the `Start` method seems less invasive to me, but I think reasonable people can differ on this one.